### ISSUE TITLE
cutf + tiny-utf-8 = ?

### ISSUE THREAD

**Issue by tapika** [DuffsDevice/tiny-utf8#27]
https://github.com/tapika/cutf

Haven't checked your library, but I've worked on something similar.

Wondering if it makes any sense to recombine bits and pieces together into one library.

**Comment by DuffsDevice** [comment]
Hi Tapika,

Thank you for asking! I just had a look at your code at I see a few similarities and a few significant differences, which suggest to me to have both libraries stay separate, which are:
- tiny-utf8 is object-oriented, cutf is procedural (or functional if you will),
- cutf targets utf16, tiny-utf8 doesn't,
- cutf8 doesn't target a pre-existing interface (AFAIK). Tiny-utf8 on the other hand implements a well-known, pre-existing, Unicode-agnostic interface and use-case (namely that of std::string).

To summarize, I think both libraries have their own unique use-case and purpose. Even though they share a lot of the algorithms, they don't share a common interface nor do I think they should.

I look forward to hearing from you.

All the Best,
Jakob
